components/Profile/MyPosts.tsx

```diff
@@ -1,24 +1,79 @@
+import { Colors } from '@/constants/Color';
+import { FontSizes, FontWeights } from '@/constants/Typography';
 import { MyPostsProps } from '@/types/components';
-import React from 'react';
+import { Collection } from '@/types/post';
+import { useActionSheet } from '@expo/react-native-action-sheet';
+import { Entypo } from '@expo/vector-icons';
+import React, { useState } from 'react';
+import { Pressable, StyleSheet, Text } from 'react-native';
 import PostList from '../Home/PostList';
 import Layout, { PADDING } from '../ui/layout/Layout';
 
 const MyPosts = ({ isMyProfile, profileInfo }: MyPostsProps) => {
+	const { showActionSheetWithOptions } = useActionSheet();
+	const [selectedCollection, setSelectedCollection] = useState<Collection>('Boards');
+
+	const options = [
+		{ value: 'Boards' as Collection, label: '마켓' },
+		{ value: 'Communities' as Collection, label: '커뮤니티' },
+	];
+
+	const selectedOption = options.find((option) => option.value === selectedCollection);
+
+	const showOptionSelect = () => {
+		const actionOptions = [...options.map(({ label }) => label), '닫기'];
+
+		showActionSheetWithOptions(
+			{
+				options: actionOptions,
+				cancelButtonIndex: actionOptions.length - 1,
+			},
+			(buttonIndex) => {
+				if (
+					buttonIndex === undefined ||
+					!options[buttonIndex] ||
+					buttonIndex === actionOptions.length - 1
+				)
+					return;
+
+				setSelectedCollection(options[buttonIndex].value);
+			},
```

Comment:
```suggestion
			(buttonIndex: number) => {
				if (
					!options[buttonIndex] ||
					buttonIndex === actionOptions.length - 1
				){
					return;
				}

				setSelectedCollection(options[buttonIndex].value);
			},
```
(가독성을 위한 제안)
1. buttonIndex가 number 타입으로 정의되어있으므로 `undefined` 확인은 생략하셔도 좋을 것 같고,
https://reactnative.dev/docs/actionsheetios#showactionsheetwithoptions
2. 코딩 스타일의 문제이긴 하지만, 한줄짜리 if문에도 `{}`를 달아주는 것이 휴먼 에러를 줄이기에 좋습니다. 😂

Reply:
<img width="1115" height="176" alt="image" src="https://github.com/user-attachments/assets/19acb941-bd8d-4fd7-8607-f0693d6d8d75" />

1. 말씀해주신 부분과 공식 문서를 읽어보고 수정해봤는데, 타입 관련 에러가 발생하더라구요. 확인해보니 제가 사용하고 있는 패키지가 `react-native` 기본 모듈이 아니라 `@expo/react-native-action-sheet` 였습니다.
이 패키지의 타입 정의에서는 `buttonIndex`가 `number | undefined`로 되어 있어서 `undefined` 체크를 포함해야할 것 같습니다. 
혹시 제가 잘못 이해한 부분이 있다면 알려주시면 감사드리겠습니다 :)
2. 이 부분 저도 동의합니다 😃

Reply:
제가 엉뚱한 문서를 보았군요! @JaneChun 님 말씀대로 undefined 체크가 필요할 것 같습니다. 😅
리뷰 검토해주셔서 감사합니다. 🙌

혹시 코드리뷰 프로세스 관련해서 공유드리면, 저는 보통 PR에서 제안을 받으면 실시간으로 커밋을 추가하고, 질의응답이 완료되거나 남은 이슈들을 다음 PR에서 다루기로 합의되면 작성자가 직접 머지하는 방식으로 진행해왔습니다.

리뷰 반영하시고 답글까지 달아주시느라 같은 코드를 여러 번 읽어보셨을 텐데 고생하셨습니다.
새 버전 무사 배포되길 기원합니다! 🎉

Reply:
코드리뷰 프로세스에 아직 익숙하지 않아 조금 헷갈렸는데, 자세히 설명해주셔서 감사합니다 🙇‍♀️
말씀해주신 대로 코드 수정 후 커밋하고 머지 진행하겠습니다 :)
바쁘실 텐데도 시간 내어 코드 리뷰해주셔서 감사드립니다!